bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+    in [Partial Signature Calculation](#partial-signature-calculation). The
+    message to be signed is
+    `MsgHash("channel_announcement", "announcement_sig", m)` where `m` is the
+    serialisation of the `channel_announcement_2` message excluding the
+    `announcement_sig` field (see the
+    [`MsgHash`](#signature-message-construction) definition).
+- otherwise:
+    - MUST NOT send the `announcement_signatures_2` message.
+- upon reconnection (once the above timing requirements have been met):
+    - MUST respond to the first `announcement_signatures_2` message with its own
+      `announcement_signatures_2` message.
+    - if it has NOT received an `announcement_signatures_2` message:
+        - SHOULD retransmit the `announcement_signatures_2` message.
+
+A recipient node:
+- if the `short_channel_id` is NOT correct:
+    - SHOULD send a `warning` and close the connection, or send an
+      `error` and fail the channel.
+- if the `partial_signature` is NOT valid as
+  per [Partial Signature Verification](#partial-signature-verification):
+    - MAY send a `warning` and close the connection, or send an
+      `error` and fail the channel.
+- if it has sent AND received a valid `announcement_signatures_2` message:
+    - SHOULD queue the `channel_announcement_2` message for its peers.
+- if it has not sent `channel_ready`:
+    - MAY send a `warning` and close the connection, or send an `error` and fail
+      the channel.
+
+### Rationale
+
+The message contains the necessary partial signature, by the sender, that the
+recipient will be able to combine with their own partial signature to construct
+the signature to put in the `channel_announcement_v2` message. Unlike the legacy
+`announcement_signatures` message, `announcement_signatures_v2` only has one
+signature field. This field is a MuSig2 partial signature which is the
+aggregation of the two signatures that the sender would have created (one for
+`bitcoin_key_x` and another for `node_ID_x`).
+
+## The `channel_announcement_2` Message
+
+This gossip message contains ownership information regarding a taproot channel. 
+It ties each on-chain Bitcoin key that makes up the taproot output key to the 
+associated Lightning node key, and vice-versa. The channel is not practically 
+usable until at least one side has announced its fee levels and expiry, using 
+`channel_update_2`.
+
+See [Taproot Channel Proof and Verification](#taproot-channel-proof-and-verification)
+for more information regarding the requirements for proving the existence of a
+channel.
+
+1. type: xxx (`channel_announcement_2`)
+2. data: (tlv_stream):
+   1. type: 0 (`announcement_sig`)
+   2. data:
+        * [`bip340_sig`:`bip340_sig`]
+   3. type: 1 (`features`)
+   4. data:
+        * [`...*byte`: `features`]
+   5. type: 2 (`chain_hash`)
+   6. data:
+        * [`chain_hash`:`chain_hash`]
+   7. type: 3 (`short_channel_id`)
+   8. data:
+        * [`short_channel_id`:`short_channel_id`]
```

Comment:
> IMO we should have an option for a channel to not include an SCID at all

since it is TLV, it can always be made optional in future.

> rather allowing a node to overcommit their balance

I think you are referring to the full on Gossip 2.0 proposal. This proposal is the more vanilla gossip 1.5 that was discussed in oakland.

Reply:
> here's a bunch of complexity to adding the negotiation of building gossip with overcommit, so I wouldn't expect to actually build that now, but there's very little complexity to allowing overcommit now

Could you expand on the details a bit?

Reply:
> Then we'd have to define some overcommit ratio (let's roll a dice or bikeshed for months), and a receiver of such a message must track the current amount of bitcoin in utxo-proven channels for each node in their graph, as well as the amount of channels which did not have a utxo proof, and if the utxo-proven channel amount is < amount of non-utxo-proven channels * the overcommit ratio the channels with the lowest (highest? whatever) fake scids are removed.

Should the overcommit amount depend on bitcoin amount in proven channels, or on number of proven channels?

Chain fees scale with number of proven channels, but not necessarily with bitcoin amount in channels.  So I think number of proven channels provides stronger spam protection.

(Sorry if this was discussed in Oakland or elsewhere, I'm new here.)

Reply:
> IMO we should have an option for a channel to not include an SCID at all, rather allowing a node to overcommit their balance by simply saying "I already have channel(s) that have utxos, let me announce more without a proof up to multiplier N by not including a proof on this channel".

This is a somewhat unsettling proposition that I'm encountering for the first time. Has this been discussed anywhere as to why people think it's a good idea? I find it unusual that we are suggesting things that resemble leverage/fractional reserve at the protocol level in Bitcoin.